master/internal/command/command_service.go

```diff
@@ -167,18 +168,26 @@ func (cs *CommandService) listByType(
 	for _, user := range reqUserIDs {
 		userIds[user] = true
 	}
+	ids := make(map[model.TaskID]bool, 0)
+	if reqIDs != nil {
+		ids = make(map[model.TaskID]bool, len(*reqIDs))
+		for _, id := range *reqIDs {
+			ids[model.TaskID(id)] = true
+		}
+	}
 
 	cmds := []*Command{}
-	for _, c := range cs.commands {
+	for id, c := range cs.commands {
 		wID := int32(c.GenericCommandSpec.Metadata.WorkspaceID)
 		username := c.Base.Owner.Username
 		userID := int32(c.Base.Owner.ID)
 		// skip if it doesn't match the requested workspaceID if any.
 		if workspaceID != 0 && workspaceID != wID {
 			continue
 		}
-		if c.taskType == cmdType && ((len(users) == 0 && len(userIds) == 0) ||
-			users[username] || userIds[userID]) {
+		if c.taskType == cmdType &&
+			((len(users) == 0 && len(userIds) == 0) || users[username] || userIds[userID]) &&
+			(len(ids) == 0 || ids[id]) {
```

Comment:
nit: this is kinda hard for me to understand

maybe we could split this up into seperate if statements, like workspaceID does